x-pack/plugins/ml/server/models/data_visualizer/data_visualizer.ts

```diff
@@ -597,23 +593,35 @@ export class DataVisualizer {
     samplerShardSize: number,
     timeFieldName: string,
     earliestMs?: number,
-    latestMs?: number
+    latestMs?: number,
+    datafeedConfig?: DatafeedOverride
   ) {
     const index = indexPatternTitle;
     const size = 0;
     const filterCriteria = buildBaseFilterCriteria(timeFieldName, earliestMs, latestMs, query);
+    const datafeedAggConfig = datafeedConfig?.aggregations ?? datafeedConfig?.aggs;
 
     // Value count aggregation faster way of checking if field exists than using
     // filter aggregation with exists query.
-    const aggs: Aggs = {};
+    const aggs: Aggs = datafeedAggConfig !== undefined ? { ...datafeedAggConfig } : {};
+
     aggregatableFields.forEach((field, i) => {
       const safeFieldName = getSafeAggregationName(field, i);
       aggs[`${safeFieldName}_count`] = {
         filter: { exists: { field } },
       };
-      aggs[`${safeFieldName}_cardinality`] = {
-        cardinality: { field },
-      };
+
+      let cardinalField: AggCardinality;
```

Comment:
suggest to rename `cardinalField` to `cardinalityField`.

Reply:
Updated here https://github.com/elastic/kibana/pull/81923/commits/d96a119950e710b043eb3c89568be037cc203ed2